### PR TITLE
Expose Inner FlightServiceClient on FlightSqlServiceClient (#3551)

### LINKED ISSUE #3551: [FlightSQL] Allow access to underlying FlightClient
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

A FlightSQL server may implement custom actions beyond what is defined by the FlightSQL protocol. However, the current [FlightSQLServiceClient](https://docs.rs/arrow-flight/31.0.0/arrow_flight/sql/client/struct.FlightSqlServiceClient.html) does not expose the action commands (`do_actions`, `list_actions`) nor allow access to the underlying `FlightServiceClient`.

**Describe the solution you'd like**

The simplest solution is to expose the underlying client by making the [mut_client()](https://github.com/apache/arrow-rs/blob/14545a42ec09782ec0371c05c01d112e0ca37604/arrow-flight/src/sql/client.rs#L131) function `pub`. Perhaps a better approach would be to add a set of `inner`/`inner_mut` functions like [`FlightClient`](https://docs.rs/arrow-flight/31.0.0/arrow_flight/client/struct.FlightClient.html#method.inner) does.

Alternatively, add a `do_action` and `list_actions` methods that forward to the underlying client.

**Describe alternatives you've considered**

The current alternative is to maintain two client instances, a `FlightSQLServiceClient` and a `FlightClient` (or `FlightServiceClient`), to be able to send custom actions.

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Closes #3551

# Rationale for this change
 

Was looking into this code as part of #3551 and realised we could simplify this

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-flight/src/sql/client.rs [hunk 1]
```diff
@@ -378,13 +378,13 @@ pub struct PreparedStatement<T> {
 
 impl PreparedStatement<Channel> {
     pub(crate) fn new(
-        client: Arc<Mutex<FlightServiceClient<Channel>>>,
+        flight_client: FlightServiceClient<Channel>,
```
Comment (tustvold) [inline arrow-flight/src/sql/client.rs:381]: Channel is cheaply cloneable, so there is no reason for the additional `Arc<Mutex>`

See https://docs.rs/tonic/latest/tonic/transport/struct.Channel.html#multiplexing-requests

**Comment by alamb** [comment]
I also have big plans for this client over the next few releases. This is a nice step forward

File: arrow-flight/src/sql/client.rs [hunk 2]
```diff
@@ -124,16 +122,18 @@ impl FlightSqlServiceClient {
...
         }
     }
 
-    fn mut_client(
-        &mut self,
-    ) -> Result<MutexGuard<FlightServiceClient<Channel>>, ArrowError> {
-        self.flight_client
-            .try_lock()
-            .map_err(|_| ArrowError::IoError("Unable to lock client".to_string()))
+    /// Return a reference to the underlying [`FlightServiceClient`]
```
Comment (alamb) [inline arrow-flight/src/sql/client.rs:129]: I also recommend an `into_inner()` function if possible

**Review by alamb — APPROVED** [review]
cc @avantgardnerio

**Review by avantgardnerio — APPROVED** [review]

**Pull request merged**